polyfill/intersection-observer.js

```diff
@@ -863,6 +863,30 @@ function getEmptyRect() {
 }
 
 
+/**
+ * Ensure that the result has all of the necessary fields of the DOMRect.
+ * Specifically this ensures that `x` and `y` fields are set.
+ *
+ * @param {?DOMRect} rect
+ * @return {?DOMRect}
+ */
+function fixDOMRect(rect) {
+  if (!rect || 'x' in rect) {
+    return rect;
+  }
+  return {
+    top: rect.top,
+    y: rect.top,
```

Comment:
Do some browsers not support the `y` property? Why is this mirroring the `top` value?

Reply:
Let me describe context here and we'll discuss what's the best path forward.

* Most of browsers use `DOMRect` structure with the shape `{x, y, top, left, width, height, bottom, right}`. The `x` and `y` simply mirror `top` and `left` when `width` and `height` are positive, which should be the case for us always.
* IE has its own `ClientRect` with the same `{top, left, width, height, bottom, right}`.
* I believe the polyfill always uses the native result of the `getBoundingClientRect()` for `Entry.boundingClientRect`. However it calculates `rootBounds` and `intersectionRect` fields.
* Consequently, I think the biggest possible source of compatibility issues is for the fields we calculate that currently always miss `x` and `y`. For instance [this computation](https://github.com/w3c/IntersectionObserver/blob/9bb05eaf228d98d807f20006cfa8de1dcc56850b/polyfill/intersection-observer.js#L806). Or event a simple [getEmptyRect](https://github.com/w3c/IntersectionObserver/blob/9bb05eaf228d98d807f20006cfa8de1dcc56850b/polyfill/intersection-observer.js#L854).
* Thus, on the platforms that support `DOMRect`, we definitely have an inconsistency, since the `boundingClientRect` has `x/y` and the other two rects do not.

I could have updated only the polyfill calculations only. But for consistency, I decided to just update "fill in" x/y in all rects returned by the `IntersectionObserverEntry`.

Reply:
Updated comments.